pkg/kubelet/kubelet.go

```diff
@@ -2567,7 +2571,8 @@ func (kl *Kubelet) HandlePodAdditions(pods []*v1.Pod) {
 		// shutting it down. If the pod hasn't started yet, we know that when
 		// the pod worker is invoked it will also avoid setting up the pod, so
 		// we simply avoid doing any work.
-		if !kl.podWorkers.IsPodTerminationRequested(pod.UID) {
+		// We also do not try to admit the pod that is already in terminated state.
+		if (!kl.podWorkers.IsPodTerminationRequested(pod.UID) && !isPodStateTerminated(pod)) {
```

Comment:
nit: can use `IsPodPhaseTerminal` - https://github.com/kubernetes/kubernetes/blob/6ac20677c781e72346ccc56daafb161d6750cbbd/pkg/api/v1/pod/util.go#L317-L319

Reply:
thanks, yes, this is better